src/docs/admin/ProActiveAdminGuide.adoc

```diff
@@ -2308,6 +2312,163 @@ In case of PAM users, the login file should not be modified as password authenti
 NOTE: Similarly to the LDAP authentication, there is a fallback mechanism for login authentication, with the difference that it is always activated.
 A user defined in the ProActive login file always has precedence over the same user defined on the Linux system.
 
+[[_keycloak]]
+=== Keycloak-based access control
+https://www.keycloak.org/[Keycloak] is an access control solution that allows single sign-on with identity and access management aimed at modern applications and services.
+ProActive is able to connect to an existing Keycloak server, and perform user authentication and authorization against it.
+
+ProActive uses the OpenID Connect (OIDC) protocol to communicate with Keycloak and acquire an access tokens for a given user.
+It further parses the obtained access token and extracts from it the user information, including the user role.
+Based on the user role, ProActive determines the set of privileges and authorizations to be granted to the user.
+
+In order to enact ProActive access control against a Keycloak server, two steps must be accomplished: (i) deploying and configuring a Keycloak server and (ii) configuring ProActive to connect to the Keycloak server and perform access control against it.
+
+==== Configuring Keycloak server
+
+WARNING: The configuration steps mentioned below provide an example of the basic steps needed to configure ProActive access control against Keycloak.
+For advanced configurations, please refer to the https://www.keycloak.org/documentation[Keycloak documentation].
+
+===== Pre-requisites
+1. A Keycloak server must be up and in operational mode so ProActive can interact with it.
+2. The administrator of the Keycloak should select a security realm (or create a new one if needed).
+
+===== Configuring a new Keycloak client for ProActive
+The administrator of the Keycloak have to configure a new client for ProActive in the chosen realm.
```

Comment:
The Keycloak server's administrator have to

Reply:
Done.